### PR TITLE
[SandboxIR][NFC] Create a DEF_CONST() macro in SandboxIRValues.def

### PR DESCRIPTION
**Pull request by vporpo**
This helps with Constant::classof().

### CODE REVIEW

File: llvm/include/llvm/SandboxIR/SandboxIR.h [hunk 1]
```diff
@@ -529,9 +534,15 @@ class Constant : public sandboxir::User {
...
   /// For isa/dyn_cast.
   static bool classof(const sandboxir::Value *From) {
-    return From->getSubclassID() == ClassID::Constant ||
-           From->getSubclassID() == ClassID::ConstantInt ||
-           From->getSubclassID() == ClassID::Function;
+    switch (From->getSubclassID()) {
+#define DEF_CONST(ID, CLASS) case ClassID::ID:
+#include "llvm/SandboxIR/SandboxIRValues.def"
+      return true;
+    case ClassID::Function:
```
Comment (tmsri) [inline llvm/include/llvm/SandboxIR/SandboxIR.h:541]: DEF_CONST(Function, Function) instead?

Comment (vporpo) [reply llvm/include/llvm/SandboxIR/SandboxIR.h:541]: Hmm I am not sure what you mean. Do you suggest replacing `case ClassID::Function:` with a `DEF_CONST` ?

Comment (tmsri) [reply llvm/include/llvm/SandboxIR/SandboxIR.h:541]: Yes, I understand you don't want to add it to SandboxIRValues.def, so just add it here?

Comment (tmsri) [reply llvm/include/llvm/SandboxIR/SandboxIR.h:541]: Include the .def file after the DEF_CONST(Function, Function) so that the undef is taken care of?

**Comment by tmsri** [comment]
LGTM

**Review by tmsri — APPROVED** [review]

**Pull request merged**